### PR TITLE
lmdb: cherry-pick Debian patch to provide a soname

### PR DESCRIPTION
**Pull request by xnox**
Without a soname, package splits; linking; SCA dependencies do not work.

Add "0" soname to lmdb by cherry-picking patch from debian.

Also fix the pc file, as lots of variables got interpreted prior to writing them out. Possibly it would have been a better idea to simply commit .pc file in the overlayd dir.

Once all that is in place, rebuild reverse dependencies to get correct linking with liblmdb.so.0 and runtime dependencies.

This makes postmap be executable without undefined symbols:

before:
```
postmap lmdb:/etc/postfix/aliases
postmap: fatal: load_library_symbols: dlopen failure loading /usr/lib/postfix/postfix-lmdb.so: /usr/lib/postfix/postfix-lmdb.so: undefined symbol: mdb_drop
```

after
```
 ldd /usr/lib/postfix/postfix-lmdb.so 
	linux-vdso.so.1 (0x00007ffd80c9a000)
	liblmdb.so.0 => /usr/lib/liblmdb.so.0 (0x00007c60f87b2000)
	libc.so.6 => /lib/libc.so.6 (0x00007c60f85bd000)
	/lib64/ld-linux-x86-64.so.2 (0x00007c60f87d4000)
```

### CODE REVIEW

File: lmdb.yaml [hunk 1]
```diff
@@ -23,21 +22,25 @@ pipeline:
...
       expected-commit: ce201088de95d26fc0da36ba805bf2ddc2ba74ff
 
+  - uses: patch
+    with:
+      patches: 0001-Debian-quirks-to-make-shared-library-versioned.patch
+
   - runs: |
       cd libraries/liblmdb
 
       cat > lmdb.pc << EOF
```
Comment (smoser) [inline lmdb.yaml:32]: fwiw, if you just use `cat >lmbd.pc << "EOF"` this would work.

i think i'd prefer that to more shell escaping.

Comment (xnox) [reply lmdb.yaml:32]: TIL!

Comment (xnox) [reply lmdb.yaml:32]: >  If any part of word is quoted, the delimiter is the result of quote removal on word, and the lines in the here-document are not expanded. If word is unquoted, all lines of the here-document are subjected to parameter expansion, command substitution, and arithmetic expansion, the character sequence \newline is ignored, and ‘\’ must be used to quote the characters ‘\’, ‘$’, and ‘`’.

from https://www.gnu.org/software/bash/manual/bash.html#Here-Documents

**Review by smoser — COMMENTED** [review]
looks good. thank you.

To satisfy curiosity, I went looking for some background on this.

Upstream Issue  [7794](https://bugs.openldap.org/show_bug.cgi?id=7794) [2014-01-29] from a fedora maintainer seems to be the first request for adding a version to lmdb.  I'm guessing all distros are just carrying a patch now.  

It is not clear to me if [8034](https://bugs.openldap.org/show_bug.cgi?id=8034#c1) is out of date now or not, but there a maintainer says:

> Just a note:
>
> LMDB should not be pulled separately from OpenLDAP.  I.e., only the bundled 
version of LMDB should be used with a given version of OpenLDAP.

Oh well.
Thank you, @xnox

File: lmdb.yaml (hunk 1 above)
Comment (smoser) [reply lmdb.yaml]: shell!

**Review by smoser — APPROVED** [review]

**Review by smoser — APPROVED** [review]

**Pull request merged**